src/components/fields/Email/Settings.tsx

```diff
@@ -0,0 +1,28 @@
+import { ISettingsProps } from "@src/components/fields/types";
+import { TextField, Button } from "@mui/material";
+
+export default function Settings({ onChange, config }: ISettingsProps) {
+
+    const copyStandardRegex = () => {
+      navigator.clipboard.writeText('^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+.[a-zA-z]{2,3}$');
```

Comment:
```suggestion
      onChange("validationRegex")("^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+.[a-zA-z]{2,3}$");
```

> As I wrote in a [previous comment](https://github.com/rowyio/rowy/pull/820#issuecomment-1264966127):
> 
> > it works when I manually update the regex (such as adding a space at the end) then saving.
> 
> > it’s only set when there is an input to the text field, so it has to be typed in first.
> 
> This is because you call `onChange` when the user types in the text field only. You manually change the value of the input element in the DOM, which doesn’t update the React state.
> 
> So when the user clicks the button, it updates the rendered input but doesn’t save in the column config, so it will never validate unless if they type in the input _after_ clicking the button.

Reply:
Hi Sidney
Can you please refer to some docs or something.
While I understood what you are trying to convey but being unable to implement the feature.

Reply:
I’ve gone ahead and merged this PR with some changes to make it work. An updated contribution guide will be published shortly, which will help for more complex contributions. [Here’s a draft link.](https://docs-git-v3-rowy.vercel.app/contributing/frontend-architecture)

For this feature:
1. The `Settings` component takes `onChange` and `config` as props. `config` is the current column config in the table’s schema. `onChange` will update the config in local state.
2. A parent component renders this `Settings` component and an “Update” button that will save the updated config stored in local state to the database.
3. Separately, the `TextField` component has its own `onChange` prop, where you can pass a function that will be called whenever the user types in that text field.
4. You set this to update the local state using `onChange` from `Settings`. So the user can type their own regex and it will be saved when they click “Update”.
5. When you created the “Use standard Regex” button, you did not use `onChange` from `Settings`, but rather changed the rendered input directly (using a ref). This means you did not “inform” React, so the local state did not change.
    - You can learn more about this in the React docs. We use [controlled components](https://reactjs.org/docs/forms.html#controlled-components) but you treated it like an [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html). The source of truth is in the local state (in the parent component).
6. When the user clicks “Update”, the parent component saves the local state. But since it was never updated, the standard regex is never saved to the database, so there is no validation.

The way to fix it is to use the `onChange` prop from `Settings` for the button, which you already did for the `TextField`.